### PR TITLE
Favor list connected to db

### PR DESCRIPTION
**Pull request by RootofalleviI**

### CODE REVIEW

File: app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java [hunk 1]
```diff
@@ -18,98 +18,97 @@
...
     Location loc = mGpsTracker.getLocation();
-
     Favor favor = new Favor(title, desc, null, loc, 0);
     FavorUtil.getSingleInstance().postFavor(favor);
+
+    // Save the favor to local favorList
+    ((MainActivity) Objects.requireNonNull(getActivity())).activeFavorArrayList.add(favor);
+
+    // Show confirmation and minimize keyboard
+    showSnackbar(getString(R.string.favor_request_success_msg));
```
Comment (huajian1069) [inline app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java]: could be replaced by commonTools

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -34,8 +34,10 @@
     <string name="favor_confirm_button">Request Favor</string>
     <string-array name="favor_list_spinner">
         <item>Active</item>
-        <item>Past</item>
+        <item>Archived</item>
     </string-array>
+    <string name="favor_no_active_favor">You have no active favor  （・∀・）</string>
```
Comment (danielpgp1012) [inline app/src/main/res/values/strings.xml]: Can you replace the（・∀・）please?

Comment (RootofalleviI) [reply app/src/main/res/values/strings.xml]: Emoji removed

File: app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java (hunk 1 above)
Comment (RootofalleviI) [reply app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java]: Replaced

File: app/src/main/java/ch/epfl/favo/view/tabs/FavorPage.java [hunk 3]
```diff
@@ -26,10 +30,26 @@
...
   }
 
+  @Override
+  public void onCreate(Bundle bundle) {
+    super.onCreate(bundle);
+
+    // Extract two arrayLists from the main activity
+    MainActivity activity = (MainActivity) Objects.requireNonNull(getActivity());
+    activeFavors = activity.activeFavorArrayList;
+    archivedFavors = activity.archivedFavorArrayList;
```
Comment (Gondolav) [inline app/src/main/java/ch/epfl/favo/view/tabs/FavorPage.java]: Remember to replace these accesses with getters ;)

**Pull request merged**